server/src/main/java/org/opensearch/cluster/metadata/ResourceLimitGroup.java

```diff
@@ -0,0 +1,288 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.metadata;
+
+import org.opensearch.cluster.AbstractDiffable;
+import org.opensearch.cluster.Diff;
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.core.ParseField;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ConstructingObjectParser;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+
+/**
+ * Class to define the ResourceLimitGroup schema
+ * {
+ *     "name": "analytics",
+ *     "resourceLimits": [
+ *          {
+ *              "resourceName": "jvm",
+ *              "value": 0.4
+ *          }
+ *     ]
+ * }
+ */
+@ExperimentalApi
+public class ResourceLimitGroup extends AbstractDiffable<ResourceLimitGroup> implements ToXContentObject {
```

Comment:
should we have an id field to refer a sandbox internally with the id. The name can be a user-facing entity.

Reply:
We should have an `isDeleted` field or something that denotes it has been deleted and in the `drain` mode only, i.e only the already running searches are left to run. Without this field, we cannot drop an incoming request when a ResourceLimitGroup is in drain mode.

Reply:
Actually, I dont think we need this as I'm guessing the delete API will remove it from the clusterState, so it wont be ever returned as a value here `activeResourceLimitGroups = new ArrayList<>(clusterService.state().metadata().resourceLimitGroups().values());`

In that case, where are we verifying the user provided ResourceLimitGroup is a valid ResourceLimitGroup or not ?